tools/hlslpm/github.py

```diff
@@ -0,0 +1,144 @@
+from enum import Enum
+from dataclasses import dataclass, field
+import os
+import sys
+from typing import List, Optional
+import requests
+from datetime import datetime, date
+import json
+
+from issue import Category, Issue
+
+
+class GH:
+    def __init__(self):
+        self.accessToken = get_pat()
+
+    def graphql(self, query: str, variables={}):
+        response = requests.post("https://api.github.com/graphql",
+                                 json={'query': query, 'variables': variables},
+                                 headers={'Authorization': f'Bearer {self.accessToken}',
+                                          'Content-Type': 'application/json'})
+
+        response.raise_for_status()
+
+        return response.json()
+
+    def project_items_summary(self):
+        query = read_file("gql/project_item_summary.gql")
+
+        pageInfo = {'endCursor': None, 'hasNextPage': True}
+
+        while pageInfo['hasNextPage']:
+            response = self.graphql(query, {'after': pageInfo['endCursor']})
+
+            items = maybe_get(response, 'data', 'organization',
+                              'projectV2', 'items')
+            for node in items['nodes']:
+                yield Issue(
+                    item_id=node['id'],
+                    item_updatedAt=to_datetime(maybe_get(node, 'updatedAt')),
+                    category=Category(
+                        maybe_get(node, 'category', 'name')),
+                    target_date=to_date(maybe_get(node, 'target', 'date')),
+                    issue_id=maybe_get(node, 'content', 'id'),
+                    issue_resourcePath=maybe_get(
+                        node, 'content', 'resourcePath'),
+                    issue_updatedAt=to_datetime(maybe_get(node, 'content', 'updatedAt')))
+
+            pageInfo = maybe_get(items, 'pageInfo')
+
+    def get_issues(self, issues: List[Issue]):
```

Comment:
I like the use of type annotations here. I think it really helps with documentation of the function. I think it would also be useful to add the return types (using the `->` syntax) for these functions with complex results.

Reply:
Added some.